nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/main/java/org/apache/nifi/excel/RowIterator.java

```diff
@@ -129,4 +151,37 @@ private boolean hasExhaustedRows() {
         }
         return exhausted;
     }
+
+    private boolean isXSSFExcelFile(InputStream in, final String password) {
+        final FileMagic fileType;
+        try {
+            fileType = FileMagic.valueOf(in);
+        } catch (final IOException e) {
+            throw new ProcessException("Failed to determine file type", e);
+        }
+
+        if (fileType == FileMagic.OOXML) {
+            // Unencrypted XLSX file
+            return true;
+        }
+        if (fileType != FileMagic.OLE2) {
+            // Not an Excel file
+            return false;
+        }
+        if (password == null) {
+            // If there is no password on the OLE2 file, then it's an XLS file. Otherwise, the type must be further
+            // investigated.
+            return false;
+        }
+
+        DirectoryNode root;
+        try {
+            root = new POIFSFileSystem(new NonCloseableInputStream(in)).getRoot();
+            in.reset();
```

Comment:
It looks like the `reset()` call should be moved to a `finally` block in case of exceptions.

Reply:
@exceptionfactory Just want to clarify since `reset` can also throw an `IOException` the `finally` would have to also have a try catch block.

Reply:
Yes, that is correct.